### ISSUE TITLE
Complex numbers

### ISSUE THREAD

**Issue by fleimgruber** [costerwi/dwdatareader#67]
For a channel with these settings (from DEWE GUI):
```
Name: U_1p_per (Complex)
Description: -
Unit: V
ID: PP1 (Power analysis)
Sampling: Asynchronous
Rate: 1043.6 Hz
Expected rate: 1000.0 Hz
Data structure: Scalar
Data type: Complex (single)
```
dwdatareader is not able to find a matching key in the `dewe_file`. Are complex numbers supported?

**Comment by fleimgruber** [comment]
I searched in the official DWDataReader C library and found `struct DWComplex` in one of the header files. Will look into it.

**Comment by costerwi** [comment]
Sorry for the delay! Do you have an idea to return complex data when it is available?

**Comment by fleimgruber** [comment]
Never mind!

As mentioned in https://github.com/costerwi/dwdatareader/issues/67#issuecomment-2419737863 I see two main approaches:

1. Add a mirrored API for complex numbers (following the C lib API).
2. Find a way to add complex numbers to the existing API.

I expect approach 1 to be straight forward, but at the cost of maintaining and exposing more APIs to the users. Approach 2 does not have these downsides, but I am not sure about the best way to check for and integrate complex numbers with the current implementation, e.g. at https://github.com/costerwi/dwdatareader/blob/master/dwdatareader/__init__.py#L394. I have not yet looked into the C lib behavior either. So I guess we should agree on one of the mentioned approaches and I would then go ahead and experiment with it a bit in a PR.

**Comment by costerwi** [comment]
Yes, I agree that approach 2 would be more convenient for everyone.

**Comment by fleimgruber** [comment]
Thanks for confirming. So next question then would be: how are users expected to refer to channels with complex type? This might depend on complex channel naming in the DEWE file. I have the feeling that users will need to somehow specify "complex" somewhere regardless for it to work.

**Comment by costerwi** [comment]
Do you have a small example file you can attach? I haven't looked but maybe the complex channels are just the imaginary component of a corresponding real channel of the same name. Python should be able to combine both components when they are available.

**Comment by fleimgruber** [comment]
Just to close the communication loop: I plan to provide an example file and an approach for access (following 1. from https://github.com/costerwi/dwdatareader/issues/67#issuecomment-2453424921) that I implemented locally.

**Comment by fleimgruber** [comment]
@Bojan023 thanks for the ping and the suggestion, will have a look!
